price-feeder/oracle/oracle.go

```diff
@@ -228,18 +243,24 @@ func (o *Oracle) SetPrices(ctx context.Context, acceptList oracletypes.DenomList
 		}
 	}
 
-	filteredProviderPrices, err := o.filterDeviations(providerPrices)
+	tvwapPrices, err := ComputeTVWAP(providerCandles)
 	if err != nil {
 		return err
 	}
 
-	vwapPrices, err := ComputeVWAP(filteredProviderPrices)
-	if err != nil {
-		return err
+	if len(tvwapPrices) < 1 {
```

Comment:
what does this conditional mean? That we have an empty set?

Reply:
Yup - mimicking what terra does here. If for some reason we haven't received candles for the last three minutes (e.g. if the websockets receive stale data), try using the last price.

Reply:
Can we just do a == 0 check? I think that's slightly cleaner.